### DISCUSSION TITLE
Help for choosing a suitable size for the PML for high wavelengths

### DISCUSSION THREAD

**Discussion by aayam2** [NanoComp/meep#2860] [Q&A]
Hello!

I am trying to simulate the reflection spectrum of a material from 110um to 3000um. Here, the lower limit of the wavelength assigns an upper limit to my resolution, and the upper limit of the wavelength assigns a lower limit for the PML size (half the maximum wavelength according to the docs for having no boundary reflections at all). However, having a PML of that size would make my box size too large, and the resolution determined by the minimum wavelength would render the simulation computationally costly and time consuming.

Any suggestions or alternate methods to solve this problem?

**Comment by gummyequation** [comment]
Hello,
Have you considered the information in the documentation referring to scalability of Maxwell's equations in the "Units in Meep" section? If I understand correctly, you can scale down all the parameters (cell size, lattice constant, wavelength, geometry, etc.) by some factor and as long as you're consistent it should yield the same results but reduced by a multiple. For the analysis, you have to remember to scale everything back up if you want to use the results. For example, say you're plotting a band diagram, you can either scale up all the frequency data.
Here is a quote from the documentation which I linked below. 
![image](https://github.com/user-attachments/assets/5f4b91fb-7cb4-49e2-9ad9-46bf9cef7cfa)
https://meep.readthedocs.io/en/latest/Introduction/#units-in-meep

**Comment by stevengj** [comment]
I would use different simulations to cover different portions of the bandwidth. Trying to cover 30x in wavelength with a single simulation will be costly.
